### PR TITLE
[Bug][Engine Server] fix checkpoint stuck error

### PR DESCRIPTION
**Pull request by EricJoy2048**

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java [hunk 1]
```diff
@@ -170,9 +170,19 @@ public CompletableFuture<Void> releaseResource(long jobId, SlotProfile profile)
...
     public boolean slotActiveCheck(SlotProfile profile) {
-        return registerWorker.values().stream()
-            .flatMap(workerProfile -> Arrays.stream(workerProfile.getAssignedSlots()))
-            .anyMatch(s -> s.getSlotID() == profile.getSlotID());
+        boolean active;
+        if (registerWorker.containsKey(profile.getWorker())) {
+            active = Arrays.stream(registerWorker.get(profile.getWorker()).getAssignedSlots())
+                .allMatch(s -> s.getSlotID() == profile.getSlotID());
+        } else {
+            active = false;
```
Comment (TaoZex) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java]: else is redundant

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java [hunk 2]
```diff
@@ -170,9 +170,19 @@ public CompletableFuture<Void> releaseResource(long jobId, SlotProfile profile)
...
+        }
+        if (!active) {
+            LOGGER.info("received slot active check failed profile: " + profile);
+        } else {
+            LOGGER.fine("received slot active check success. profile: " + profile);
```
Comment (TaoZex) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java]: remove .

Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java]: > remove .

Thanks, Would it be better add `.` after `received slot active check failed` ?

Comment (TaoZex) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/AbstractResourceManager.java]: I agree to your suggestion.

**Review by TaoZex — DISMISSED** [review]

**Review by ashulin — DISMISSED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by ic4y — APPROVED** [review]
LGTM

**Pull request merged**